data/test/macros/wml_unit_test_macros.cfg

```diff
@@ -198,14 +222,24 @@ numerical_equals#endarg
             variable=i
             start=0
             step=1
-            end="$({WEAPONS}-1)"
+            end="$($strikes.length-1)"
             [do]
-                [test_do_attack_by_id]
-                    attacker={ID1}
-                    defender={ID2}
-                    weapon=$i
-                    resupply_attacks_left=5
-                [/test_do_attack_by_id]
+                [if]
+                    [have_unit]
+                        id={ID1}
+                    [/have_unit]
+                    [have_unit]
+                        id={ID2}
+                    [/have_unit]
```

Comment:
It's the caller's responsibility to know that both units are on the map at this point - if it doesn't know, then that's a bug in the test. Think about bisecting a failure of such a test.

Reply:
This is needed for the `berserk_negative_value` test for example since otherwise it will try execute an attack with the side 1 leader's second weapon even though it has died, which results in an error being logged and the unit test result being `BROKE_STRICT_TEST_PASS` instead of `PASS`.

Reply:
Updated.